src/convertToObject.js

```diff
@@ -7,6 +7,21 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString.split(';').reduce((resultObj, segment) => {
+    const cleanSegment = segment.trim();
+
+    if (!cleanSegment) {
+      return resultObj;
+    }
+
+    const [key, data] = cleanSegment.split(':').map((part) => part.trim());
+
+    if (key && data) {
+      resultObj[key] = data;
+    }
+
+    return resultObj;
```

Comment:
The 'reduce' method is used correctly here to transform the string into an object. However, make sure to maintain consistent indentation when chaining methods for better readability. In this case, the code is readable, but keep this in mind for future reference.